Modules/timemodule.c

```diff
@@ -418,37 +436,41 @@ tmtotuple(struct tm *p
     return v;
 }
 
-/* Parse arg tuple that can contain an optional float-or-None value;
-   format needs to be "|O:name".
-   Returns non-zero on success (parallels PyArg_ParseTuple).
-*/
 static int
-parse_time_t_args(PyObject *args, const char *format, time_t *pwhen)
+object_to_time_t(PyObject *obj, time_t *when)
 {
-    PyObject *ot = NULL;
-    time_t whent;
-
-    if (!PyArg_ParseTuple(args, format, &ot))
-        return 0;
-    if (ot == NULL || ot == Py_None) {
-        whent = time(NULL);
+    if (obj == NULL || obj == Py_None) {
+        *when = time(NULL);
     }
-    else {
-        if (_PyTime_ObjectToTime_t(ot, &whent, _PyTime_ROUND_FLOOR) == -1)
-            return 0;
+    else if (_PyTime_ObjectToTime_t(obj, when, _PyTime_ROUND_FLOOR) == -1) {
+        return 0;
     }
-    *pwhen = whent;
     return 1;
 }
 
+/*[clinic input]
+time.gmtime
+
+    seconds: object = NULL
```

Comment:
Maybe use a converter?
```suggestion
    seconds as when: object(converter="object_to_time_t", type="time_t") = None
```
Or introduce a named converter?

Note also that you can use `None` as a default value for better signature.